sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidCorrelateUnnestRel.java

```diff
@@ -0,0 +1,211 @@
+{
+  // This may be needed for the explain plan later
+  // private static final TableDataSource DUMMY_DATA_SOURCE = new TableDataSource("__unnest__");
+  private final PartialDruidQuery partialQuery;
+  private final PlannerConfig plannerConfig;
+  private final LogicalCorrelate logicalCorrelate;
+  private final DataSource baseDataSource;
+  private final DruidQueryRel druidQueryRel;
+  private final Filter baseFilter;
+  private DruidUnnestDatasourceRel unnestDatasourceRel;
+
+  public DruidCorrelateUnnestRel(
+      RelOptCluster cluster,
+      RelTraitSet traitSet,
+      LogicalCorrelate logicalCorrelateRel,
+      PartialDruidQuery partialQuery,
+      DruidQueryRel druidQueryRel,
+      DruidUnnestDatasourceRel unnestDatasourceRel,
+      Filter baseFilter,
+      PlannerContext plannerContext
+  )
+  {
+    super(cluster, traitSet, plannerContext);
+    this.logicalCorrelate = logicalCorrelateRel;
+    this.partialQuery = partialQuery;
+    this.plannerConfig = plannerContext.getPlannerConfig();
+    this.druidQueryRel = druidQueryRel;
+    this.baseDataSource = druidQueryRel.getDruidTable().getDataSource();
+    this.unnestDatasourceRel = unnestDatasourceRel;
+    this.baseFilter = baseFilter;
+  }
+
+
+  @Nullable
+  @Override
+  public PartialDruidQuery getPartialDruidQuery()
+  {
+    return partialQuery;
+  }
+
+  @Override
+  public DruidCorrelateUnnestRel withPartialQuery(PartialDruidQuery newQueryBuilder)
+  {
+    return new DruidCorrelateUnnestRel(
+        getCluster(),
+        getTraitSet().plusAll(newQueryBuilder.getRelTraits()),
+        logicalCorrelate,
+        newQueryBuilder,
+        druidQueryRel,
+        unnestDatasourceRel,
+        baseFilter,
+        getPlannerContext()
+    );
+  }
+
+  @Override
+  public DruidQuery toDruidQuery(boolean finalizeAggregations)
+  {
+    final RowSignature rowSignature = RowSignatures.fromRelDataType(
+        logicalCorrelate.getRowType().getFieldNames(),
+        logicalCorrelate.getRowType()
+    );
+
+    final DruidQuery leftQuery = Preconditions.checkNotNull((druidQueryRel).toDruidQuery(false), "leftQuery");
+    final DataSource leftDataSource;
+
+    if (DruidJoinQueryRel.computeLeftRequiresSubquery(druidQueryRel)) {
+      leftDataSource = new QueryDataSource(leftQuery.getQuery());
+    } else {
+      leftDataSource = leftQuery.getDataSource();
+    }
+
+    final DruidExpression expression = Expressions.toDruidExpression(
+        getPlannerContext(),
+        rowSignature,
+        unnestDatasourceRel.getUnnestProject().getProjects().get(0)
+    );
+
+    LogicalProject unnestProject = LogicalProject.create(
+        this,
+        ImmutableList.of(unnestDatasourceRel.getUnnestProject()
+                                            .getProjects()
+                                            .get(0)),
+        unnestDatasourceRel.getUnnestProject().getRowType()
+    );
+
+    String dimensionToUnnest;
```

Comment:
nit: final? also suggest using `inputToUnnest` or something since the input doesn't have to be a dimension

Reply:
Done